rag_apps/hybrid_rag_system/src/document_processor.py

```
@@ -0,0 +1,70 @@
+
+def chunk_text(text: str) -> list[str]:
+    splitter = RecursiveCharacterTextSplitter(
+        chunk_size=CHUNK_SIZE,
+        chunk_overlap=CHUNK_OVERLAP,
+        separators=["\n\n", "\n", ". ", "! ", "? ", " ", ""],
+    )
+    return [c for c in splitter.split_text(text) if c.strip()]
+
+
+def compute_hash(content: str) -> str:
+    return hashlib.md5(content.encode()).hexdigest()
```

Ruff (0.15.12)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)